src/System.Text.RegularExpressions/src/System/Text/ValueStringBuilder.Reverse.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text
+{
+    internal ref partial struct ValueStringBuilder
+    {
+        public void AppendReversed(ReadOnlySpan<char> value)
+        {
+            Span<char> span = AppendSpan(value.Length);
+            value.CopyTo(span);
```

Comment:
You should copy it, one character at a time, so that it lands in memory reversed.